src/mjolnir/pbfgraphparser.cc

```diff
@@ -1859,7 +1868,7 @@ struct graph_callback : public OSMPBF::Callback {
 
     // We need to set a data processing flag so we need to
     // process in pbfgraphparser instead of lua because of config option use_rest_area
-    if (use_rest_area_ && service_ == "rest_area") {
+    if (use_rest_area_ && service_ == "rest_area" && way_.use() != Use::kConstruction) {
```

Comment:
this is kind of unfortunate, there are several places where we need to detect if its under construction and then if it is we avoid modifying the use or the access. it doesnt seem like there is any way around having to remember to check this for every one. i was going to suggest doing it inside of the setters in the way class so its less spread out over the code but its stil the same amound of checking unfortunately..

Reply:
agree with you, ideally all logic about modifying `use` or `access` should be concentrated in one place. Right now it's not true. So, let me check if it can be moved into "way_setters"

Reply:
yep i came to the same conclusion, i was really just kind of sad that we had to put this all over the place and then remember to put it the next time we start parsing something new that has to do with access/use.